sdk/messaging/azservicebus/internal/namespace.go

```diff
@@ -169,10 +165,19 @@ func NamespaceWithAzureEnvironment(namespaceName, environmentName string) Namesp
 	}
 }
 
-// NamespaceWithTokenProvider sets the token provider on the namespace
-func NamespaceWithTokenProvider(provider auth.TokenProvider) NamespaceOption {
+// NamespacesWithTokenCredential sets the token provider on the namespace
+func NamespacesWithTokenCredential(namespaceName string, tokenCredential azcore.TokenCredential) NamespaceOption {
 	return func(ns *Namespace) error {
-		ns.TokenProvider = provider
+		ns.TokenProvider = newTokenProviderWithTokenCredential(tokenCredential)
+
+		parts := strings.SplitN(namespaceName, ".", 2)
```

Comment:
Not sure what the format is supposed to be.  Should this go in the doc comment for this constructor?

Reply:
Updated. I realized after you wrote this that I probably shouldn't allow just a barename without a suffix.